### PR TITLE
Misc 3

### PR DESCRIPTION
**Pull request by jschueller**
- Fix duplicated symbols from submodules FittingTest, HypothesisTest, etc being also exposed at the root of the module eg ot.ChiSquared in addition to ot.FittingTest.ChiSquared
- Cleanup SpecFunc: drop duplicate/unused/unnecessary (wrt c++11 or Python.math) functions & doc
- expose IdentityFunction in Python
- unpin sphinx, requires sphincontrib-jquery for the search, see #2480

### CODE REVIEW

File: lib/test/t_InverseMarginalTransformationHessian_std.cxx
Comment (josephmure) [inline lib/test/t_InverseMarginalTransformationHessian_std.cxx:1]: Why are these tests unused? Do they have no purpose?

File: lib/test/t_MarginalDistribution_std.cxx [hunk 1]
```diff
@@ -103,14 +95,14 @@ int main(int, char *[])
...
+    // fullprint << "pdf gradient     =" << PDFgr << std::endl;
+    // Point CDFgr = distribution.computeCDFGradient( point );
+    // fullprint << "cdf gradient     =" << CDFgr << std::endl;
     // Confidence regions
-    Scalar threshold;
-    fullprint << "Minimum volume interval=" << distribution.computeMinimumVolumeIntervalWithMarginalProbability(0.95, threshold) << std::endl;
-    fullprint << "threshold=" << threshold << std::endl;
+//     Scalar threshold;
+//     fullprint << "Minimum volume interval=" << distribution.computeMinimumVolumeIntervalWithMarginalProbability(0.95, threshold) << std::endl;
+//     fullprint << "threshold=" << threshold << std::endl;
```
Comment (josephmure) [inline lib/test/t_MarginalDistribution_std.cxx:105]: Pourquoi supprimer tous ces tests?

File: lib/test/t_InverseMarginalTransformationHessian_std.cxx
Comment (jschueller) [reply lib/test/t_InverseMarginalTransformationHessian_std.cxx:1]: looks the correspoind class was removed

File: lib/test/t_MarginalDistribution_std.cxx (hunk 1 above)
Comment (jschueller) [reply lib/test/t_MarginalDistribution_std.cxx]: not yielding same results on win32, but could not make it work

Comment (josephmure) [reply lib/test/t_MarginalDistribution_std.cxx]: Maybe replace them with assert-based tests?

Comment (josephmure) [reply lib/test/t_MarginalDistribution_std.cxx]: Could you please unresolve this? I would like it to remain visible for the next technical committee meeting.

File: python/src/Brent_doc.i.in [hunk 2]
```diff
@@ -29,6 +24,12 @@ Notes
 The Brent solver is a mix of Bisection, Secant and inverse quadratic
 interpolation.
 
+If the function :math:`f` is continuous, the Brent solver will
+converge towards a root of the equation :math:`function(x) = value` in
+:math:`[infPoint, supPoint]`. If not, it will converge towards either a root or
+a discontinuity point of :math:`f` on :math:`[infPoint, supPoint]`.
+Bisection guarantees a convergence.
```
Comment (josephmure) [inline python/src/Brent_doc.i.in:31]:
```suggestion
Bisection guarantees convergence.
```

File: python/src/SolverImplementation_doc.i.in [hunk 3]
```diff
@@ -167,3 +167,31 @@ nEval : int
...
+
+Parameters
+----------
+function : :class:`~openturns.Function`
+    The function of the equation :math:`function(x) = value` to be solved in the
+    interval :math:`[infPoint, supPoint]`.
+value : float
+    The value to which the function must be equal.
+infPoint, supPoint : float
+    Lower and upper bounds of the interval definition of the variable :math:`x`.
```
Comment (josephmure) [inline python/src/SolverImplementation_doc.i.in:184]: What does "the interval definition of x" mean?

File: python/src/SpecFunc_doc.i.in [hunk 4]
```diff
@@ -97,6 +97,25 @@ result : float"
 
 // ---------------------------------------------------------------------
 
+%feature("docstring") OT::SpecFunc::BesselKDerivative
+"Derivative of modified second kind Bessel function of order :math:`\nu`.
```
Comment (josephmure) [inline python/src/SpecFunc_doc.i.in:101]:
```suggestion
"Derivative of the modified second kind Bessel function of order :math:`\nu`.
```

File: python/src/SpecFunc_doc.i.in [hunk 5]
```diff
@@ -120,6 +139,28 @@ result : float"
 
 // ---------------------------------------------------------------------
 
+%feature("docstring") OT::SpecFunc::DeltaLogBesselI10
+"Difference between the logarithms of Bessel I1 and I0.
+
+.. math::
+
+    \forall x \in \Rset, \quad
+    DeltaLogBesselI10(x) = \log(\mathrm{I}_1(x)) - \log(\mathrm{I}_0(x))
```
Comment (josephmure) [inline python/src/SpecFunc_doc.i.in:148]: Maybe use \mathrm for "DeltaLogBesselI10"?

File: python/src/SpecFunc_doc.i.in [hunk 6]
```diff
@@ -345,6 +386,28 @@ result : float or complex (same as z)"
 
 // ---------------------------------------------------------------------
 
+%feature("docstring") OT::SpecFunc::Factorial
+"Factorial function.
+
+.. math::
+
+    \forall n \in \Nset, \quad
+    Factorial(n) = n!
```
Comment (josephmure) [inline python/src/SpecFunc_doc.i.in:395]: Same

File: python/src/SolverImplementation_doc.i.in (hunk 3 above)
Comment (jschueller) [reply python/src/SolverImplementation_doc.i.in]: yes, lets use "range" instead

File: python/src/StudentCopulaFactory_doc.i.in [hunk 7]
```diff
@@ -27,6 +27,32 @@ DistributionFactory, StudentCopula
...
+sample : 2-d sequence of float
+    Data.
+param : sequence of float
+    The parameters of the :class:`~openturns.StudentCopula`.
+
+Returns
+-------
+dist : :class:`~openturns.Uniform`
+    The estimated distribution as a :class:`openturns.StudentCopula`.
+    In the first usage, the default :class:`openturns.StudentCopula` distribution is built."
```
Comment (josephmure) [inline python/src/StudentCopulaFactory_doc.i.in:52]:
```suggestion
    The estimated distribution as a :class:`~openturns.StudentCopula`.
    In the first usage, the default :class:`~openturns.StudentCopula` distribution is built."
```

File: python/src/Study_doc.i.in [hunk 8]
```diff
@@ -227,13 +227,13 @@ hasObject : bool
 
 // ---------------------------------------------------------------------
 
-%feature("docstring") OT::Study::printLabels
-"Print all the names of the objects stored in the study.
+%feature("docstring") OT::Study::getLabels
+"Accessor to the labels of the objects stored.
```
Comment (josephmure) [inline python/src/Study_doc.i.in:231]:
```suggestion
"Accessor to the labels of the stored objects.
```

File: python/test/t_Waarts_noisy_lsf.py [hunk 9]
```diff
@@ -10,23 +11,12 @@
...
+formulas = [
+    "X1 + 2*X2 + 2*X3 + X4 - 5*X5 - 5*X6 +0.001*(sin(100*X1)+sin(100*X2)+sin(100*X3)+sin(100*X4)+sin(100*X5)+sin(100*X6))"]
 
-formulas = ot.Description(outputFunction.getSize())
-formulas[
-    0
-] = "X1 + 2*X2 + 2*X3 + X4 - 5*X5 - 5*X6 +0.001*(sin(100*X1)+sin(100*X2)+sin(100*X3)+sin(100*X4)+sin(100*X5)+sin(100*X6))"
-
-EtatLimite = ot.SymbolicFunction(inputFunction, outputFunction, formulas)
+EtatLimite = ot.SymbolicFunction(inputFunction, formulas)
```
Comment (josephmure) [inline python/test/t_Waarts_noisy_lsf.py:19]: Why is this name in French?

File: python/test/t_Waarts_noisy_lsf.py [hunk 10]
```diff
@@ -56,61 +46,15 @@
...
-
-# Create a sixth marginal : distribution 1D
-marginal = ot.LogNormal(mean[5], sigma[5], BorneInf, ot.LogNormal.MUSIGMA)
-marginal.setName("Sixth")
-component[0] = "Six"
-marginal.setDescription(component)
-# Fill the sixth marginal of aCollection
-aCollection.add(ot.Distribution(marginal, "Sixth"))
+for i in range(len(mean)):
+    marginal = ot.LogNormalMuSigma(mean[i], sigma[i], BorneInf).getDistribution()
```
Comment (josephmure) [inline python/test/t_Waarts_noisy_lsf.py:50]: Here too.

**Review by josephmure — CHANGES_REQUESTED** [review]

**Comment by jschueller** [comment]
@regislebrun @josephmure could we review this ?

**Review by sofianehaddad — APPROVED** [review]

**Pull request merged**